website/blog/2022-07-04-aluminium.md

```diff
@@ -0,0 +1,467 @@
+
+## Release notes now shown in VS Code
+
+If you are using Visual Studio Code you should see this message right after the
+update of the Metals extension has finished. 🚀
+
+Starting with the current release we will be showing release notes inside VS
+Code when the version update completes. This way users will know whenever the
+extension was updated and it will not just silently update in the background.
+
+Release notes will be shown only once after the update, but can be viewed again
+by executing `Metals: Show release notes`
+
+This was contributed by [kpodsiad](https://github.com/kpodsiad), thanks for the
+continuous contributions!
+
+## Improvements to Bloop Java home automation
+
+From the
+[last version of Metals](https://scalameta.org/metals/blog/2022/06/03/aluminium#automatically-setup-java-home-for-the-bloop-build-server)
+we introduced an automated setup of Java home for your Bloop build server, which
+is used to compile your code under the hood and is the default build server to
+be used.
+
+Your Java Home would be added when starting Metals, before starting the build
+server, and would also suggest an update when changing the versions in the
+Metals settings. However, this caused a couple of issues:
+
+1. In some cases Metals would suggest to change your Java home even if it hasn't
+   changed and it wasn't possible to dismiss the notification forever.
+2. In case your Java home changed and you missed the notification (or the
+   notification was not shown), Bloop would not start if the old java home was
+   removed. That especially caused issues with usage of Java with Nix.
+3. Metals would suggest to update your ~/.bloop/bloop.json file even if it was
+   readonly.
+
+To help with that we implemented a number of fixes:
+
+1. You can dismiss the notification forever if Metals mistakenly suggests
+   updating your Bloop configuration and the overall false positives have been
+   reduced.
+2. Metals will check before starting if your Java home is correct, and if it's
+   not it will replace it with the current one used by Metals.
+3. Metals will not show any notifications if it cannot change the file.
+
+If you encounter any further issues, you can take a look at Metals output and it
+should show logs like these:
+
+```
+2022.07.01 16:14:41 INFO  Bloop uses /usr/lib/jvm/openjdk17 defined at /home/tgodzik/.bloop/bloop.json
+2022.07.01 16:14:41 INFO  Bloop currently uses settings: -Xmx3G
+```
+
+You can use them to debug what is happening and report an issue.
+
+## Allow to run all scalafix rules on a file
+
+Up until the current release Metals would only run scalafix for
+`organize imports` and wouldn't allow to use any other rules. Starting with this
+release you should be able to apply all currently defined
+`rules in .scalafix.conf` you require by running `metals.scalafix-run` command.
+
+By default it's bound to `alt` + `ctrl` + `shift` + `o` shortcut in Visual
+Studio Code.
+
+![scalafix-run](https://i.imgur.com/raVjlRX.gif)
```

Comment:
Can you record this GIF with screencast mode? (`Developer: Toggle Screencast Mode`)

Reply:
Changed, thanks for the suggestion!